backend/src/main/java/gov/cdc/usds/simplereport/api/apiuser/UserAccountCreationController.java

```diff
@@ -2,27 +2,48 @@
 
 import static gov.cdc.usds.simplereport.config.WebConfiguration.USER_ACCOUNT_REQUEST;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.smartystreets.api.exceptions.BadRequestException;
 import gov.cdc.usds.simplereport.api.model.errors.InvalidActivationLinkException;
 import gov.cdc.usds.simplereport.api.model.errors.OktaAuthenticationFailureException;
+import gov.cdc.usds.simplereport.api.model.useraccountcreation.ActivateAccountRequest;
 import gov.cdc.usds.simplereport.api.model.useraccountcreation.ActivateSecurityKeyRequest;
 import gov.cdc.usds.simplereport.api.model.useraccountcreation.EnrollMfaRequest;
 import gov.cdc.usds.simplereport.api.model.useraccountcreation.FactorAndQrCode;
+import gov.cdc.usds.simplereport.api.model.useraccountcreation.SetPasswordRequest;
 import gov.cdc.usds.simplereport.api.model.useraccountcreation.SetRecoveryQuestionRequest;
-import gov.cdc.usds.simplereport.api.model.useraccountcreation.UserAccountCreationRequest;
+import gov.cdc.usds.simplereport.api.model.useraccountcreation.UserAccountStatus;
 import gov.cdc.usds.simplereport.idp.authentication.OktaAuthentication;
+import javax.annotation.Nullable;
 import javax.annotation.PostConstruct;
 import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpSession;
+import javax.validation.Valid;
 import org.json.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.SessionAttribute;
 
-/** Controller used for user account creation. */
+/**
+ * Controller used for user account creation.
+ *
+ * <p>There's an implicit flow in this controller. The expectation is that the user is first
+ * activated, then their password is set, then recovery questions, and finally they are enrolled in
+ * and activate an MFA option. A sample endpoint flow is as follows: /initialize, /set-password,
+ * /set-recovery-questions, /enroll-sms-mfa, /verify-activation-passcode.
+ *
+ * <p>Okta has an internal state machine that keeps track of the user's status, and will reject
+ * requests if they're not the next step in the state. For example, attempting to enroll a user in
+ * MFA before they've set their password will generate an Okta error, because the user is still in a
```

Comment:
Yep! It'll most likely be re-thrown as an `OktaAuthenticationFailureException`, since I'm catching all the raw Okta exceptions in `LiveOktaAuthentication`. (Their high-level exception is a `ResourceException`, so I'm just try/catching that in most cases)